### PR TITLE
rust/sip: register parser for tcp v13

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3351

Describe changes:
-  PDU frame is created for each SIP message when it is successfully parsed.

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1538
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17109

**Review by victorjulien — APPROVED** [review]

**Review by jasonish — APPROVED** [review]

File: rust/src/sip/sip.rs [hunk 1]
```diff
@@ -147,20 +148,76 @@ impl SIPState {
...
+        if input.is_empty() {
+            return AppLayerResult::ok();
+        }
+
+        let mut start = input;
+        while !start.is_empty() {
+            match sip_parse_request(start) {
+                Ok((rem, request)) => {
+                    let pdu_len = (start.len() - rem.len()) as i64;
+                    let _f = Frame::new(
```
Comment (victorjulien) [inline rust/src/sip/sip.rs:164]: the frame should be created earlier, and not depend on the parser returning ok. It should be created as soon as we start parsing the data, and then be updated with the length when it's complete

File: rust/src/sip/sip.rs [hunk 2]
```diff
@@ -112,27 +109,31 @@ impl SIPState {
...
-            &stream_slice,
-            input,
-            input.len() as i64,
-            SIPFrameType::Pdu as u8,
-        );
-        SCLogDebug!("ts: pdu {:?}", _pdu);
 
         match sip_parse_request(input) {
             Ok((_, request)) => {
+                let _pdu = Frame::new(
```
Comment (victorjulien) [inline rust/src/sip/sip.rs:127]: frame should not depend on the parser returning ok

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: rust/src/sip/sip.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/sip/sip.rs:164]: Sorry for the confusion, I was the one who asked this.

What should the length be updated to when the parser errors (incomplete or not) ?

Why does it have to be so ?
On the S-V test, a frame was created before returning incomplete, and suricata did not match on it... 
So, why should we create a frame if we cannot match on it ? (or how can we match on it)

File: rust/src/sip/sip.rs (hunk 2 above)
Comment (catenacyber) [reply rust/src/sip/sip.rs:127]: see above

File: rust/src/sip/sip.rs (hunk 1 above)
Comment (victorjulien) [reply rust/src/sip/sip.rs:164]: Indeed, the telnet code seems to do it well.

We need this so we can start matching immediately for large frames. Also if we have malformed data we can still do frame based content matching.

Comment (catenacyber) [reply rust/src/sip/sip.rs:164]: > We need this so we can start matching immediately for large frames. Also if we have malformed data we can still do frame based content matching.

This does not seem to work cf https://github.com/OISF/suricata/pull/10037 CI is green with SV https://github.com/OISF/suricata-verify/pull/1538 where there is only one alert for https://github.com/OISF/suricata-verify/pull/1538/files#diff-efc7fbd99eb9389e40b2f9f16aa9eca8d58a80ea88324756b8a5d4b476cf69cfR18 `alert sip any any -> any any (flow:to_server; frame:pdu; content:"REGISTER"; startswith; sid:2;)` after the frame is created on packet 6 (and no alert for the frame created on packet 4, as can be checked adding a `bsize` keyword to the rule)

**Comment by glongo** [comment]
Replaced by https://github.com/OISF/suricata/pull/10098

**Pull request closed without merging**